spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java

```diff
@@ -68,17 +86,28 @@ public ModuleLauncher(ModuleResolver moduleResolver) {
 	 * <code>&lt;groupId&gt;:&lt;artifactId&gt;[:&lt;extension&gt;[:&lt;classifier&gt;]]:&lt;version&gt;</code>
 	 *
 	 * @param moduleLaunchRequests a list of modules with their arguments
+	 * @param aggregate whether the modules should be aggregated at launch
+	 * @param parentArgs a list of arguments for the whole aggregate
 	 */
-	public void launch(List<ModuleLaunchRequest> moduleLaunchRequests) {
+	public void launch(List<ModuleLaunchRequest> moduleLaunchRequests, boolean aggregate, String parentArgs[]) {
 		List<ModuleLaunchRequest> reversed = new ArrayList<>(moduleLaunchRequests);
 		Collections.reverse(reversed);
-		for (ModuleLaunchRequest moduleLaunchRequest : reversed) {
-			String module = moduleLaunchRequest.getModule();
-			moduleLaunchRequest.addArgument("spring.jmx.default-domain", module.replace("/", ".").replace(":", "."));
-			launchModule(module, toArgArray(moduleLaunchRequest.getArguments()));
+		if (moduleLaunchRequests.size() == 1 || !aggregate) {
+			launchModulesIndividually(moduleLaunchRequests);
```

Comment:
would be nice if these methods were named consistently, e.g. `launchIndividualModules` and `launchAggregatedModules`